internal/grpc/services/usershareprovider/usershareprovider.go

```diff
@@ -497,40 +506,156 @@ func (s *service) GetReceivedShare(ctx context.Context, req *collaboration.GetRe
 }
 
 func (s *service) UpdateReceivedShare(ctx context.Context, req *collaboration.UpdateReceivedShareRequest) (*collaboration.UpdateReceivedShareResponse, error) {
-
-	if req.Share == nil {
-		return &collaboration.UpdateReceivedShareResponse{
-			Status: status.NewInvalid(ctx, "updating requires a received share object"),
-		}, nil
-	}
-	if req.Share.Share == nil {
-		return &collaboration.UpdateReceivedShareResponse{
-			Status: status.NewInvalid(ctx, "share missing"),
-		}, nil
-	}
-	if req.Share.Share.Id == nil {
-		return &collaboration.UpdateReceivedShareResponse{
-			Status: status.NewInvalid(ctx, "share id missing"),
-		}, nil
-	}
-	if req.Share.Share.Id.OpaqueId == "" {
+	if req.GetShare().GetShare().GetId().GetOpaqueId() == "" {
 		return &collaboration.UpdateReceivedShareResponse{
 			Status: status.NewInvalid(ctx, "share id empty"),
 		}, nil
 	}
 
+	isStateTransitionShareAccepted := slices.Contains(req.GetUpdateMask().GetPaths(), _fieldMaskPathState) && req.GetShare().GetState() == collaboration.ShareState_SHARE_STATE_ACCEPTED
+	if isStateTransitionShareAccepted {
+		gatewayClient, err := s.gatewaySelector.Next()
+		if err != nil {
+			return nil, err
+		}
+
+		receivedShare, err := gatewayClient.GetReceivedShare(ctx, &collaboration.GetReceivedShareRequest{
+			Ref: &collaboration.ShareReference{
+				Spec: &collaboration.ShareReference_Id{
+					Id: req.GetShare().GetShare().GetId(),
+				},
+			},
+		})
+		switch {
+		case err != nil:
+			fallthrough
+		case receivedShare.GetStatus().GetCode() != rpc.Code_CODE_OK:
+			return &collaboration.UpdateReceivedShareResponse{
+				Status: receivedShare.GetStatus(),
+			}, err
+		}
+
+		resourceStat, err := gatewayClient.Stat(ctx, &provider.StatRequest{
+			Ref: &provider.Reference{
+				ResourceId: receivedShare.GetShare().GetShare().GetResourceId(),
+			},
+		})
+		switch {
+		case err != nil:
+			fallthrough
+		case resourceStat.GetStatus().GetCode() != rpc.Code_CODE_OK:
+			return &collaboration.UpdateReceivedShareResponse{
+				Status: resourceStat.GetStatus(),
+			}, err
+		}
+
+		// handle mount point related updates
+		{
+			// find a suitable mount point
+			var requestedMountpoint string
+			switch {
+			case slices.Contains(req.GetUpdateMask().GetPaths(), _fieldMaskPathMountPoint) && req.GetShare().GetMountPoint().GetPath() != "":
+				requestedMountpoint = req.GetShare().GetMountPoint().GetPath()
```

Comment:
I am still not completely sure about the semantics when the incoming request actually contains a MountPoint. I think in that case we should not do the game with the prefixed number. But rather be explicit. I.e. if the mountpoint is already taken, make the request fail and return an appropriate Status.

Reply:
I’m with you, but that means clients have to implement that a user can change the mount point name (I would prefer that too).

but that needs some time, maybe we should merge it that way, talk to the clients and then change it to error in the case of explicit conflicts

Reply:
> I’m with you, but that means clients have to implement that a user can change the mount point name (I would prefer that too).

Hm, I am not sure I get that. Clients (as in CS3 clients) are already able to change the mount point name, aren't they?

The fact that our currently used clients at the higher-level (every thing that comes in via libregraph e.g.) don't make use of that does not change that. But in that case they shouldn't send a mountpoint at all.

Reply:
Alternatively I'd be find if we'd just not allow UpdateReceivedShare Request with a `_fieldMaskPathMountPoint` at all for now.